packages/react/src/index.js

```diff
@@ -10,49 +10,50 @@ const createCss = (init) => {
 	const sheet = createCoreCss(init)
 
 	return assign(sheet, {
-		/** Returns a React component. */
+		/** Returns a React forwarded ref object. */
 		styled: (
 			/** Type of component. */
 			...inits
 		) => {
-			const defaultType = inits.map((init) => (Object(init).type ? init.type : init)).find((init) => init) || 'span'
+			const defaultType = inits.map((init) => (Object(init).stitchesType ? init.stitchesType : init)).find((init) => init) || 'span'
 			const composition = sheet.css(...inits.filter((init) => $$composers in Object(init) || (init && typeof init === 'object' && !init.$$typeof)))
 
-			/** Returns a React element. */
-			return Object.setPrototypeOf(
-				{
-					render(
-						/** Props used to determine the expression of the current component. */
-						initProps,
-						ref,
-					) {
-						// express the component, extracting `props`, `as` & `ref`
-						const {
-							props: { as: type = defaultType, ...props },
-							...expressedProps // eslint-disable-line no-unused-vars
-						} = composition(initProps)
+			/** This is a React forwarded ref object, with a few extra Stitches properties */
+			return {
+				/** The render function is the React component on the forwarded ref object, it returns a React element. */
+				render(
+					/** Props used to determine the expression of the current component. */
+					initProps,
+					ref,
+				) {
+					// express the component, extracting `props`, `as` & `ref`
+					const {
+						props: { as: type = defaultType, ...props },
+						...expressedProps // eslint-disable-line no-unused-vars
+					} = composition(initProps)
 
-						/** React element. */
-						return { constructor: undefined, $$typeof: $$typeofElement, props, ref, type, __v: 0 }
-					},
-					$$typeof: $$typeofForward,
-					[$$composers]: composition[$$composers],
-					[Symbol.toPrimitive]() {
-						return composition.selector
-					},
-					toString() {
-						return composition.selector
-					},
-					get className() {
-						return composition.className
-					},
-					get selector() {
-						return composition.selector
-					},
-					type: defaultType,
+					/** React element. */
+					return { $$typeof: $$typeofElement, key: null, props, ref, type }
```

Comment:
I would recommend replacing this with `React.createElement(type, Object.assign(props, ref))`, or use JSX like `<Type {...props} ref={ref} />`. This way the `_owner` property can be added to the element object for a better dev experience, and it will avoid bugs like `key="undefined"` (or something worse) should React internals change in the future (which I don't believe would require a semver breaking change as manually creating the element object is not part of the React api).

Reply:
This would probably also help with supporting Preact #454